### DISCUSSION TITLE
Relevance tuning with weights

### DISCUSSION THREAD

**Discussion by sebestyenhunor** [Jeroen-G/Explorer#231] [Q&A]
Is it possible to set weights to the fields? Like the search_fields weights in the [elastic docs](https://www.elastic.co/guide/en/app-search/current/relevance-tuning-guide.html).

**Comment by Jeroen-G** [comment]
I don't really understand what they are, so maybe.

**Comment by sebestyenhunor** [comment]
If you don't know about it, Explorer can't do it, I guess. Similar to boost, only these weights are calculated on fields, not on specific values.  Sort by relevance for a product search should have the following field weights, and elasticsearch API waits this structure:
```
{
  "search_fields": {
    "title": {
      "weight": 10
    },
    "description": {
      "weight": 0
    },
    "sku": {
      "weight": 10
    },
    "attributes": {
      "weight": 5
    }
  },
  "query": "yellow T-shirt"
}
```

Is there any way to add 'search_fields' to the query using explorer?
Thank you!

**Reply by Jeroen-G** [reply]
If it is part of the query you can write a new Syntax and pass that as the search query (see the docs on advanced queries). Should be very doable!

**Reply by sebestyenhunor** [reply]
Unfortunately, this isn't part of the query, but is next to the query in the request body. Any method or idea for this case?
Thank you?

**Reply by Jeroen-G** [reply]
How would the ES query look like that you can execute via Kibana?

**Comment by sebestyenhunor** [comment]
Sorry, this kind of weighting is available in the latest ES version (8<), I was not attentive.
So I must use boosting, the only problem is that boost with 0 doesn't set 0 score for description field. I have four fields (title, sku, attributes, description), all of them are text type, and all of them are used by autocomplete, but description matches should not occur at the top of sort by relevance. I also need that the position of the search term in the index field should boost the relevant score, e.g. 'AWSOME product' should have higher score than 'second AWSOME product' for 'AWSO' search term, since occurs earlier in the string.
Do you have any examples or ideas for this?
Thank you!
